src/Security/Handler/AclSecurityHandler.php

```diff
@@ -186,6 +187,7 @@ public function findObjectAcls(\Traversable $oids, array $sids = [])
 
     public function addObjectOwner(AclInterface $acl, ?UserSecurityIdentity $securityIdentity = null)
     {
+        \assert($acl instanceof MutableAclInterface);
```

Comment:
In these situations, I would prefer a check with an explicit error, something like
```
if (!$acl instanceof MutableAclInterface) {
    throw \InvalidArgumentException('$acl must implement MutableAclInterface');
}
```
in order to help the developer.